lib/msal-core/src/ClientInfo.ts

```diff
@@ -63,4 +63,28 @@ export class ClientInfo {
             throw ClientAuthError.createClientInfoDecodingError(e);
         }
     }
+
+    static stripPolicyFromUid(uid: string, authority: string): string {
+        const uidSegments = uid.split("-");
+        const urlSegments = authority.split("/").reverse();
+        let policy = ""
+
+        if (!StringUtils.isEmpty(urlSegments[0])) {
+            policy = urlSegments[0];
+        } else if (urlSegments.length > 1) {
+            policy = urlSegments[1];
+        }
```

Comment:
Could you add some comments here describing what parsing is going on here?

Reply:
I also wonder whether it would be better to have a getPolicy on the authority object and pass the policy string into this function.

Reply:
Parsing the authority string is not the long term solution and this part of the logic will be replaced by a separate "policy" field on the ClientInfo object once the work is done by the server. Given that this is temporary and we no longer have a B2CAuthority class I'd rather leave the changes confined to ClientInfo for now.

Will add comments however.